lib/codegen/ProofEvent.cpp

```diff
@@ -38,34 +38,26 @@ llvm::CallInst *proof_event::emit_serialize_term(
   auto *i8_ptr_ty = llvm::Type::getInt8PtrTy(ctx_);
   auto *i1_ty = llvm::Type::getInt1Ty(ctx_);
 
-  if (cat.cat == sort_category::Symbol || cat.cat == sort_category::Variable) {
-    auto *block_ty = getvalue_type({sort_category::Symbol, 0}, module_);
-
-    auto *func_ty = llvm::FunctionType::get(
-        void_ty, {i8_ptr_ty, block_ty, i8_ptr_ty, i1_ty}, false);
-
-    auto *serialize
-        = get_or_insert_function(module_, "serialize_term_to_file", func_ty);
+  llvm::ConstantInt *construct_k_term_inj = nullptr;
```

Comment:
Rather than having this variable be initialized to `nullptr` and reassigned, perhaps consider a single call to `ConstantInt::getBool`?